docs/concepts/serialization.md

```diff
@@ -465,6 +484,153 @@ annotated as `<SomeType>`, and type-checkers like mypy will treat the attribute
 But when serializing, the field will be serialized as though the type hint for the field was `Any`, which is where the
 name comes from.
 
+#### `serialize_as_any` runtime setting
+
+The `serialize_as_any` runtime setting can be used to serialize model data with or without duck typed serialization behavior.
+`serialize_as_any` can be passed as a keyword argument to the `model_dump()` and `model_dump_json` methods of `BaseModel`s and `RootModel`s. It can also be passed as a keyword argument to the `dump_python()` and `dump_json()` methods of `TypeAdapter`s.
+
+If `serialize_as_any` is set to `True`, the model will be serialized using duck typed serialization behavior,
+which means that the model will ignore the schema and instead ask the object itself how it should be serialized.
+In particular, this means that when model subclasses are serialized, fields present in the subclass but not in
+the original schema will be included.
+
+If `serialize_as_any` is set to `False` (which is the default), the model will be serialized using the schema,
+which means that fields present in a subclass but not in the original schema will be ignored.
+
+For example:
+
+```py
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    name: str
+
+
+class UserLogin(User):
+    password: str
+
+
+class OuterModel(BaseModel):
+    user1: User
+    user2: User
+
+
+user = UserLogin(name='pydantic', password='password')
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=True)) # (1)!
+"""
+{
+    'user1': {'name': 'pydantic', 'password': 'password'},
+    'user2': {'name': 'pydantic', 'password': 'password'},
+}
+"""
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=False)) # (2)!
+"""
+{
+    'user1': {'name': 'pydantic'},
+    'user2': {'name': 'pydantic'},
+}
+"""
+```
+
+1. With `serialize_as_any` set to `True`, the result matches that of V1.
+2. With `serialize_as_any` set to `False` (the V2 default), fields present on the subclass,
```

Comment:
I feel like it's maybe worth explaining the benefit of this, which may not be clear to users reading these docs — sometimes, you want to make sure that no matter what fields might have been added in subclasses, the serialized object will only have the fields listed in the original type definition. This can be useful if you might add something like `password: str` field in a subclass.